glusterd2/middleware/auth.go

```diff
@@ -31,11 +31,27 @@ func getAuthSecret(issuer string) string {
 	return ""
 }
 
+//isRestAuthRequired return false for few URL which doesn't require authentication
+func isRestAuthRequired(url string) bool {
+	if !gdctx.RESTAPIAuthEnabled {
+		return false
+	}
+	switch url {
+	case "/ping":
+		fallthrough
+	case "/endpoints":
+		return false
+	default:
+		return true
+	}
+
+}
+
 // Auth is a middleware which authenticates HTTP requests
 func Auth(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// If Auth disabled Return as is
-		if !gdctx.RESTAPIAuthEnabled {
+		if !isRestAuthRequired(r.URL.String()) {
```

Comment:
Suggestion:

`isRestAuthRequired` takes URL as argument but can decide just based on `gdctx.RESTAPIAuthEnabled`. This is slightly less clean. I'd prefer to separate the checks for slightly better readability.

```go
if (auth not enabled) || (auth not required for URL) {
    pass through
}
```

In this case, we don't even bother checking URL if auth is disabled

Reply:
agreed, looks better. made changes please check